File: Interviews/GFG/random-problems/Medium/Count-of-Subarrays/Count-of-Subarrays.py
class Solution:
    def countSubarray(self, arr, n, k):
        ans, maxx = 0, [0, -1]
        for idx, i in enumerate(arr):
            if i > k:
                maxx = [i, idx]
            ans += maxx[-1] + 1
        return ans

File: Interviews/GFG/random-problems/Medium/Count-of-Subarrays/Count-of-Subarrays.cpp
class Solution {
   public:
    // #define ll long long
    ll countSubarray(int arr[], int n, int k) {
        ll ans = 0;
        vector<int> maxx = {0, -1};
        for (int i = 0; i < n; i++) {
            if (arr[i] > k) {
                maxx = {arr[i], i};
            }
            ans += maxx[1] + 1;
        }

        return ans;
    }
};